packages/cipherstash-proxy/src/postgresql/messages/close.rs

```diff
@@ -0,0 +1,144 @@
+use crate::error::{Error, ProtocolError};
+use crate::postgresql::protocol::BytesMutReadString;
+use crate::{SIZE_I32, SIZE_U8};
+
+use bytes::{Buf, BufMut, BytesMut};
+use std::convert::TryFrom;
+use std::ffi::CString;
+use std::io::Cursor;
+
+use super::target::Target;
+use super::{FrontendCode, Name};
+
+///
+/// Close b'C' (Frontend) message.
+///
+/// See: <https://www.postgresql.org/docs/current/protocol-message-formats.html>
+///
+///     Byte1('C')
+///     Identifies the message as a Close command.
+///
+///     Int32
+///     Length of message contents in bytes, including self.
+///
+///     Byte1
+///     'S' to close a prepared statement; or 'P' to close a portal.
+///
+///     String
+///     The name of the prepared statement or portal to close (an empty string selects the unnamed prepared statement or portal).
+
+#[derive(Debug, Clone)]
+pub(crate) struct Close {
+    pub target: Target,
+    pub name: Name,
+}
+
+impl TryFrom<&BytesMut> for Close {
+    type Error = Error;
+
+    fn try_from(bytes: &BytesMut) -> Result<Close, Self::Error> {
+        let mut cursor = Cursor::new(bytes);
+        let code = cursor.get_u8();
+
+        if FrontendCode::from(code) != FrontendCode::Close {
+            return Err(ProtocolError::UnexpectedMessageCode {
+                expected: FrontendCode::Close.into(),
+                received: code as char,
+            }
+            .into());
+        }
+
+        let _len = cursor.get_i32(); // read and progress cursor
+        let target = cursor.get_u8();
+        let target = Target::try_from(target)?;
+        let name = cursor.read_string()?;
+        let name = Name::from(name);
+
+        Ok(Close { target, name })
+    }
+}
+
+impl TryFrom<Close> for BytesMut {
+    type Error = Error;
+
+    fn try_from(close: Close) -> Result<BytesMut, Error> {
+        let mut bytes = BytesMut::new();
+
+        let name = CString::new(close.name.as_str())?;
+        let name = name.as_bytes_with_nul();
+
+        let len = SIZE_I32 + SIZE_U8 + name.len();
```

Comment:
Rust has `std::mem::size_of<T>()` to return sizes of types in bytes.

Consider using `size_of::<i32>() + size_of::<u8>() + name.len()` which is more idiomatic than defining our own constants primitive type sizes.

Reply:
@freshtonic The values are used in a few places and are defined using size_of:

```
pub const SIZE_U8: usize = mem::size_of::<u8>();
pub const SIZE_I16: usize = mem::size_of::<i16>();
pub const SIZE_I32: usize = mem::size_of::<i32>();
```